### PR TITLE
WebExtension Messaging documentation

### PR DESCRIPTION
**Pull request by agi90**
rendered: https://agi.sferro.dev/geckoview/docs/web-extensions.html

### CODE REVIEW

**Comment by snorp** [comment]
Maybe we should put these couple examples in a subdirectory called `extensions` or similar?

**Comment by agi90** [comment]
> Maybe we should put these couple examples in a subdirectory called `extensions` or similar?

yeah that sounds good, done.

File: docs/web-extensions.md [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+    "myextension@example.com",
+    // Extra flags can be specified here
+    WebExtension.Flags.NONE);
+
+// Run the WebExtension
+runtime.registerWebExtension(extension);
+```
+
+Note that the lifetime of the WebExtension is tied with the lifetime of the
+[`GeckoRuntime`][24] instance. WebExtension will need to be registered every
```
Comment (fluffyemily) [inline docs/web-extensions.md]: `The WebExtension` or `Your WebExtension`.

**Review by fluffyemily — APPROVED** [review]
Documentation as a first class citizen. I love it. 🎉

**Comment by fluffyemily** [comment]
Closes #23

File: docs/web-extensions.md (hunk 1 above)
Comment (agi90) [reply docs/web-extensions.md]: Done! Thanks :)

**Pull request merged**